tests/src/test/scala/actionContainers/ActionContainer.scala

```diff
@@ -88,12 +88,37 @@ object ActionContainer {
   }
 
   private lazy val dockerCmd: String = {
-    val version = WhiskProperties.getProperty("whisk.version.name")
-    // Check if we are running on docker-machine env.
-    val hostStr = if (version.toLowerCase().contains("mac")) {
-      s" --host tcp://${WhiskProperties.getMainDockerEndpoint} "
+    /*
+     * The docker host is set to a provided property 'docker.host' if it's
+     * available; otherwise we check with WhiskProperties to see whether we are
+     * running on a docker-machine.
+     *
+     * IMPLICATION:  The test must EITHER have the 'docker.host' system
+     * property set OR the 'OPENWHISK_HOME' environment variable set and a
+     * valid 'whisk.properties' file generated.  The 'docker.host' system
+     * property takes precedence.
+     *
+     * WARNING:  Adding a non-docker-machine environment that contains 'mac'
+     * (i.e. 'environments/local-mac') will likely break things.
+     *
+     * The plan is to move builds to using 'gradle-docker-plugin', which know
+     * its docker socket and to have it pass the docker socket implicitly using
+     * 'systemProperty "docker.host", docker.url'.  Eventually, we will also
+     * need to handle TLS certificates here.  Again, 'gradle-docker-plugin'
+     * knows where they are; we will just add system properties to get the
+     * information onto the docker command line.
+     */
+    val dockerUrl = Option(System.getProperty("docker.host"))
+    val hostStr = if (dockerUrl.isDefined) {
+      s" --host ${dockerUrl.get} "
     } else {
-      " "
+      val version = WhiskProperties.getProperty("whisk.version.name")
+      // Check if we are running on docker-machine env.
+      if (version.toLowerCase().contains("mac")) {
```

Comment:
What do you think making the unit tests work with no “whisk.properties” present at all?
By adding a check if System.getenv("OPENWHISK_HOME")/WhiskProperties.WHISK_PROPS_FILE doesn’t exist.

Reply:
Same thing below 
Check if file doesn’t exist and onOSX the. Use a port
Or if file exist read if local and OSX then same as today use a port

Reply:
If file doesn’t exist and not OSX then don’t use a port

Reply:
Could be.  I was thinking this morning of stringing together several options and using whichever came back first with an affirmative result:

1). The ‘docker.host’ Java property (which is what I’ll be using in multi-arch builds)
2).  The ‘DOCKER_HOST’ environment variable (which is set on docker-machine systems)
3).  The ‘WhiskProperties’ logic if ‘WhiskProperties’ exists (add a try/catch) block
4).  Just run ‘docker info’ and see if it works
5).  Fail

I can foldLeft an ordered list of anonymous functions implementing each of those to make it fairly maintainable.  Of course, it’ll take us some time to test the tests...

Reply:
Yeah sounds good

Reply:
Okay.  It'll take a little time to do and test.  I'll mark this WIP for now.